http-core/src/main/scala/org/apache/pekko/http/scaladsl/model/HttpHeader.scala

```diff
@@ -99,10 +99,11 @@ object HttpHeader {
             case HeaderParser.RuleNotFound => ParsingResult.Ok(RawHeader(name, preProcessedValue), Nil)
           }
         case Failure(error) =>
-          val info = (error match {
-            case e: ParseError => parser.parseError(e)
-            case e             => parser.failure(e)
-          }).info
+          val info = error match {
+            case e: ParseError             => parser.parseError(e).info
+            case e if e.getMessage == null => ErrorInfo()
```

Comment:
I think this code should be reverted and the change should be in HeaderParser.scala instead. I have the change in #576. I created 576 to test if that approach works but I'd prefer to let you complete this PR. View 576 as a set of suggested changes for this PR.